src/main/java/com/google/cloud/genomics/dataflow/pipelines/ShardedBAMWriting.java

```diff
@@ -267,15 +309,24 @@ public HeaderInfo apply(String str) throws Exception {
           String contigStr = str.substring(0, newLinePos);
           String headerStr = str.substring(newLinePos + 1);
           return new HeaderInfo(
-              SAM_HEADER_CODEC.decode(new StringLineReader(headerStr),
+              SAM_HEADER_CODEC.decode(new StringLineReader(headerStr), 
                   "HEADER_INFO_CODER"),
               Contig.parseContigsFromCommandLine(contigStr).iterator().next());
         }
       });
 
-  static final long LOCI_PER_SHARD = 10000;
-
-  static Contig shardKeyForRead(Read read) {
+  static boolean isUnmapped(Read read) {
```

Comment:
There is isUnmapped(Position) but not isUnmapped(Read).
Now, I agree we should add one, so as usual I will file an issue to extract common code from here to utils as part of a refactoring step.